### ISSUE TITLE
Pod lib lint doesn't validate tvOS target

### ISSUE THREAD

**Issue by jeffh** [CocoaPods/CocoaPods#6593]
🌈

* [x] I've read and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

With an incorrect podspec configuration for tvOS (eg - [missing files](https://github.com/Quick/Nimble/commit/e3c1a78af33b3fa778207fcc6ba33221838c0788) in target).

```bash
pod lib lint Nimble.podspec
# ...tag repo + push tag...
pod trunk push Nimble.podspec
```

## What did you expect to happen?

I expected `pod lib lint` to fail instead of only finding it out on `pod trunk push`.

## What happened instead?

`pod lib lint` passes but than `pod trunk push` fails validation because it can't compile tvOS. Alternatively if there's another command I should be running in CI to validate a podspec across multiple targets. One usually doesn't like to find out if the podspec is valid half-way through cutting a release 😅.

### Stack

```
   CocoaPods : 1.2.1.beta.1
        Ruby : ruby 2.4.0p0 (2016-12-24 revision 57164) [x86_64-darwin16]
    RubyGems : 2.6.8
        Host : Mac OS X 10.12.4 (16E195)
       Xcode : 8.3 (8E162)
         Git : git version 2.11.0 (Apple Git-81)
Ruby lib dir : /usr/local/var/rbenv/versions/2.4.0/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 60f90f7c52833d380e182bc80989d90329eb8505
```

### Installation Source

```
Executable Path: /usr/local/var/rbenv/versions/2.4.0/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.1
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.1.2
cocoapods-try         : 1.1.0
```

## Project that demonstrates the issue

N/A

**Comment by benasher44** [comment]
@jeffh thanks for the report! This sounds like a bug. I can't guarantee I timeline for when this will be fixed, but I've tagged the issue appropriately. Since you have a working repro case, we'd certainly appreciate a PR or further investigation with details posted here if you can spare the time! Thanks again!

**Comment by paulb777** [comment]
We took a look and `pod lib lint` doesn't catch the error because it builds in place and can find the header file that happens to be in the same directory as the CwlMachBadInstructionHandler.m file that fails to build in `pod spec lint`. `pod spec lint` builds the sources from a clean temp directory so it catches the missing header problem.

In short, it is better to run `pod spec lint` instead of `pod lib lint` before `pod trunk push`.

**Comment by dnkoutso** [comment]
@paulb777 interesting...perhaps we should have the validator clean stuff for each platform.

**Comment by paulb777** [comment]
@dnkoutso Yeah, building in a clean directory would make `pod lib lint` more consistent with `pod spec lint`
